pf/tests/internal/providerbuilder/build_provider.go

```diff
@@ -36,6 +36,8 @@ type Provider struct {
 	Version        string
 	ProviderSchema schema.Schema
 	AllResources   []Resource
+
+	onConfigure func(tfsdk.Config)
```

Comment:
Nit: The `build_resource` has a slightly different naming convention for the method implementations, should we follow that: https://github.com/pulumi/pulumi-terraform-bridge/blob/67961059383e8551fd5dddddce3ac6e08cc139d7/pf/tests/internal/providerbuilder/build_resource.go#L28

Reply:
The `*Func` do something slightly different, but I can change `onConfigure func(tfsdk.Config)` to 
```go
ConfigureFunc func(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse)
```

It's a more cohesive design, but a more powerful hook.